org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/InspectorInfo.java

```diff
@@ -9,24 +9,100 @@
  */
 package org.eclipse.scout.rt.ui.html.json;
 
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import org.eclipse.scout.rt.client.ui.AbstractWidget;
 import org.eclipse.scout.rt.platform.ApplicationScoped;
 import org.eclipse.scout.rt.platform.classid.ITypeWithClassId;
+import org.eclipse.scout.rt.platform.exception.ProcessingException;
+import org.eclipse.scout.rt.platform.util.StringUtility;
 import org.eclipse.scout.rt.server.commons.servlet.UrlHints;
-import org.eclipse.scout.rt.ui.html.IUiSession;
 import org.json.JSONObject;
 
 /**
+ * Adds inspector properties ({@value #PROP_CLASS_ID}, {@value #PROP_MODEL_CLASS}) to a {@link JSONObject}.
+ *
  * @since 5.2
  */
 @ApplicationScoped
 public class InspectorInfo {
 
-  public void put(IUiSession uiSession, JSONObject json, Object model) {
-    if (model != null && UrlHints.isInspectorHint(uiSession.currentHttpRequest())) {
-      json.put("modelClass", model.getClass().getName());
-      if (model instanceof ITypeWithClassId) {
-        json.put("classId", ((ITypeWithClassId) model).classId());
-      }
+  public static final String PROP_CLASS_ID = "classId";
+  public static final String PROP_MODEL_CLASS = "modelClass";
+
+  /**
+   * Regex for a UUID as created by {@link UUID}.
+   */
+  private static final String UUID_PATTERN = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}";
+  /**
+   * Pattern for concatenated classIds as e.g. created by {@link AbstractWidget#classId()}.
+   */
+  public static final Pattern CLASS_ID_WITH_UUID_PATTERN = Pattern.compile(UUID_PATTERN + "(?:" + ITypeWithClassId.ID_CONCAT_SYMBOL + UUID_PATTERN + ")*");
+  private static final MessageDigest SHA256 = createSha256Digest();
+
+  private static MessageDigest createSha256Digest() {
+    try {
+      return MessageDigest.getInstance("SHA-256");
+    }
+    catch (NoSuchAlgorithmException e) {
+      throw new ProcessingException("Unable to create SHA-256 message digest.", e);
+    }
+  }
+
+  /**
+   * Adds inspector properties ({@value #PROP_CLASS_ID} & {@value #PROP_MODEL_CLASS}) to the given {@link JSONObject}.
+   *
+   * @param req
+   *     The {@link HttpServletRequest} to detect if the inspector is enabled
+   *     ({@link UrlHints#isInspectorHint(HttpServletRequest)}). The {@value #PROP_MODEL_CLASS} property is only
+   *     added if it is enabled. May be {@code null}.
+   * @param json
+   *     The target {@link JSONObject} that should receive the properties. May be {@code null} then this method
+   *     does nothing.
+   * @param model
+   *     The model for which the properties should be added. May be {@code null} then this method does nothing.
+   */
+  public <T> void put(HttpServletRequest req, JSONObject json, T model) {
```

Comment:
Instead of passing the http request, consider a boolean flag instead, e.g. `addModelClass`. The API feels strange right now, since we are not "putting" something on the http request, but on the json object. The http qurest is only used to determine the inspector hint.